src/NUnitFramework/framework/Internal/HookExtension.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Linq;
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// Hook Extension interface to run custom code synchronously before or after any test activity.
+    /// </summary>
+    public class HookExtension
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HookExtension"/> class.
+        /// </summary>
+        public HookExtension()
+        {
+            BeforeAnySetUps = (context, method) => { };
+            AfterAnySetUps = (context, method) => { };
+            BeforeTest = (context, testMethod) => { };
+            AfterTest = (context, testMethod) => { };
+            BeforeAnyTearDowns = (context, method) => { };
+            AfterAnyTearDowns = (context, method) => { };
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HookExtension"/> class by copying hooks from another instance.
+        /// </summary>
+        /// <param name="other">The instance of <see cref="HookExtension"/> to copy hooks from.</param>
+        public HookExtension(HookExtension other) : this()
+        {
+            other.BeforeAnySetUps?.GetInvocationList()?.ToList().ForEach(d => BeforeAnySetUps += d as SetUpTearDownHookHandler);
```

Comment:
Are you only calling `ToList` so you can call `ForEach`?
According to [Nick Chapsas  ](https://www.youtube.com/watch?v=jUZ3VKFyB-A) the `ForEach` method is 3 times slower than iterating yourself and that is if you already have a list to begin with. It is worse when you first have to allocate one. `GetInvocationList()` returns an array and can be looped over directly.

Reply:
As I expect not hundreds or thousands registered hooks I did not care about performance and instead focused on readability and compact code.
Do you think this could be an error?

Reply:
But this can be also easily done be a small helper.
But we do not need the reverse order if we got for async hooks => wait for feedback on sync or async here....